contracts/PoolManager.sol

```diff
@@ -138,14 +135,16 @@ contract PoolManager is IPoolManager, Fees, NoDelegateCall, ERC1155, IERC1155Rec
 
     /// @inheritdoc IPoolManager
     function lock(bytes calldata data) external override returns (bytes memory result) {
+        LockData lockData = LockDataLibrary.getLockData();
```

Comment:
something definitely feels a bit weird to me about using pop/push (write functionality) on a type uint256 that actually does storage writes. I wish there was a nice way to return a reference to storage here instead of recaching. Feels a bit dirty but can't think of any great solutions...

Reply:
once there's a full solc library that has the transient keyword, I think it's worth toying around with having a storage slot (unless we can call things directly on storage here with the current lib). I know it saves gas, but I'm wondering if the code cleanliness/quality is getting a little bit wonky here

Reply:
Yeah it is a bit wonky..let me know if you can think of another way, but I had to move away from a library for storage pointers because we don't yet have the `transient` keyword. 

We can rewrite the lib to use the transient keyword when there is solidity support.